visualization/app/codeCharta/codeCharta.service.ts

```diff
@@ -25,29 +25,19 @@ export class CodeChartaService {
 
 	async loadFiles(nameDataPairs: NameDataPair[]) {
 		this.fileStates = this.storeService.getState().files
-		for (const nameDataPair of nameDataPairs) {
-			try {
-				validate(nameDataPair)
-				this.addFile(nameDataPair)
-			} catch (error) {
-				if (error.error.length > 0) {
-					this.fileStates = []
-					this.recentFiles = []
-					this.storeService.dispatch(setIsLoadingFile(false))
-					await this.dialogService.showValidationErrorDialog(error)
-					break
-				}
-
-				if (error.warning.length > 0) {
-					this.addFile(nameDataPair)
-					this.addRecentFile(nameDataPair.fileName)
-					this.storeService.dispatch(setIsLoadingFile(false))
-					await this.dialogService.showValidationWarningDialog(error)
-				}
-			}
-		}
+		const fileValidationResults: CCFileValidationResult[] = []
+
+		this.getValidationResults(nameDataPairs, fileValidationResults)
 
-		if (this.fileStates.length > 0) {
+		this.storeService.dispatch(setIsLoadingFile(false))
+
+		if (
+			fileValidationResults.flatMap(validation => validation.errors).length > 0 ||
+			fileValidationResults.flatMap(validation => validation.warnings).length > 0
+		)
+			await this.dialogService.showValidationDialog(fileValidationResults)
+
+		if (this.recentFiles.length > 0) {
```

Comment:
It looks like that recentFiles is not set anymore,

Reply:
it is being set in method addFile